TLM/TLM/U/Panel/BaseUWindowPanel.cs

```diff
@@ -11,43 +13,79 @@ namespace TrafficManager.U.Panel {
             uiScaleUnbsubscriber_ = ModUI.Instance.UiScaleObservable.Subscribe(this);
+            uiTransparencyUnbsubscriber_ = ModUI.Instance.UiTransparencyObservable.Subscribe(this);
         }
 
         public UResizerConfig GetResizerConfig() {
             return resizerConfig_;
         }
 
-        public void OnResizerUpdate() { }
+        /// <summary>Called by UResizer for every control before it is to be 'resized'.</summary>
+        public virtual void OnBeforeResizerUpdate() { }
 
-        /// <summary>Called on screen resolution and UI scale change.</summary>
-        public abstract void OnRescaleRequested();
+        /// <summary>Called by UResizer for every control after it is to be 'resized'.</summary>
+        public virtual void OnAfterResizerUpdate() { }
 
         /// <summary>Invoke rescaling handler, because possibly it has the new size now.</summary>
         /// <param name="previousResolution">Previous.</param>
         /// <param name="currentResolution">New.</param>
         protected override void OnResolutionChanged(Vector2 previousResolution,
                                                     Vector2 currentResolution) {
-            this.OnRescaleRequested();
+            // Call resize on all controls and recalculate again
+            UResizer.UpdateControlRecursive(this, null);
+        }
+
+        /// <summary>
+        /// Impl. <see cref="IObserver{T}"/> for UI Scale changes.
+        /// Called from ModUI when UI scale slider in General tab was modified.
+        /// </summary>
+        /// <param name="optionsEvent">New UI scale.</param>
+        public void OnUpdate(ModUI.UIScaleNotification optionsEvent) {
+            // Call resize on all controls and recalculate again
+            UResizer.UpdateControlRecursive(this, null);
         }
 
         /// <summary>
         /// Impl. <see cref="IObserver{T}"/> for UI Scale changes.
         /// Called from ModUI when UI scale slider in General tab was modified.
         /// </summary>
-        /// <param name="uiScale">New UI scale</param>
-        public void OnUpdate(ModUI.UIScaleNotification uiScale) {
-            this.OnRescaleRequested();
+        /// <param name="uiScale">New UI scale.</param>
+        public void OnUpdate(ModUI.UITransparencyNotification optionsEvent) {
+            // incoming range: 0..100 convert to 0..1f
+            SetTransparency(optionsEvent.NewTransparency);
+        }
+
+        /// <summary>
+        /// Rewrite window color to become transparent.
+        /// NOTE: If the call has no effect, look for some other code rewriting the color after your call!
+        /// </summary>
+        /// <param name="guiTransparencyPercent">Range 0..100.</param>
+        internal void SetTransparency(float guiTransparencyPercent) {
+            Color32 modified = this.color;
+
+            // converting range 0..100f to range 0..255
+            modified.a = (byte)(255f * 0.01f * guiTransparencyPercent);
```

Comment:
I would rather rename the GUI label than modify the formula :) But yes you're right.

Reply:
If you change the GUI label you should also rename all related variables/methods/.. that have transparency in their name to opacity so that their name suggest they do what they actually do. Its going to be not a minor change and vulnerable to mistake and confusion. Why do you not want to change the formula?

Reply:
BTW just to make sure we are on the same page functionally If you set transparency slider to 100% it will be 0% transparent.

Reply:
I keep the slider as it is now, because it is named transparency in the configuration file, but internally it is inverted as per your formula.

Reply:
One problem is that default value is 100, we'll get the invisible menu.